File: StaffWebApp/models/credit.go
package models

type CreditRate struct {
	Id      string  `json:"id"`
	Name    string  `json:"name"`
	Percent float64 `json:"monthPercent"`
}

type CreditShort struct {
	Id            string     `json:"id"`
	UserId        string     `json:"userId"`
	Rate          CreditRate `json:"creditRate"`
	RemainingDebt Money      `json:"remainingDebt"`
	UnpaidDebt    Money      `json:"unpaidDebt"`
}

type CreditDetailed struct {
	CreditShort
	LinkedAccount  string `json:"payingAccountId"`
	MoneyTaken     Money  `json:"fullMoneyAmount"`
	MonthlyPayment Money  `json:"monthPayAmount"`
}

File: client-bank/backend/client-bank-backend/client-bank-backend/DTOs/Credit.cs
using client_bank_backend.DTOs;
using CreditApplication.Models.Dtos;

namespace CreditApplication.Models
{
    public class Credit : BaseEntity
    {
        public Guid CreditRateId { get; set; }
        public CreditRate CreditRate{ get; set; }

        public Guid UserId {  get; set; }
        public Guid PayingAccountId { get; set; }

        public Money FullMoneyAmount { get; set; }
        public Money MonthPayAmount { get; set; }

        public Money RemainingDebt { get; set; }

        public Money UnpaidDebt { get; set; }
    }
}
